ReactAndroid/src/main/java/com/facebook/react/uimanager/NativeViewHierarchyManager.java

```diff
@@ -383,7 +390,48 @@ public void manageChildren(
                       viewsToAdd,
                       tagsToDelete));
         }
-        dropView(viewToDestroy);
+
+        if (mLayoutAnimator.shouldAnimateLayout(viewToDestroy)) {
+          disableUserInteractions(viewToDestroy);
+          mLayoutAnimator.deleteView(viewToDestroy, new Callback() {
+            @Override
+            public void invoke(Object... args) {
+              ViewGroup parent = (ViewGroup)viewToDestroy.getParent();
+              if (parent != null) {
+                viewManager.removeView(parent, viewToDestroy);
+              }
+              dropView(viewToDestroy);
+            }
+          });
+        } else {
+          dropView(viewToDestroy);
+        }
+      }
+    }
+  }
+
+  /**
+   * Checks if a view's tag is contained in the tagsToDelete array.
+   */
+  private boolean isViewDeleted(View view, int[] tagsToDelete) {
+    int viewTag = view.getId();
+    for (int tag : tagsToDelete) {
+      if (tag == viewTag) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Disables user interactions for a view and all it's subviews.
+   */
+  private void disableUserInteractions(View view) {
```

Comment:
This should not live in NativeViewHierarchyManager imo, but in the animation class itself.